rust/headless-client/src/standalone.rs

```diff
@@ -154,7 +157,9 @@ pub fn run_only_headless_client() -> Result<()> {
     platform::setup_before_connlib()?;
     let args = ConnectArgs {
         url,
-        sockets: Sockets::new(),
+        sockets: Sockets::new(Box::new(crate::udp_socket_factory)),
+        udp_socket_factory: Arc::new(crate::udp_socket_factory),
+        tcp_socket_factory: Arc::new(crate::tcp_socket_factory),
```

Comment:
Looking at the different places `ConnectArgs` is used, is there any case where `sockets` couldn't just be implemented as `Sockets::new(Box::new(args.tcp_socket_factory))` inside the `connect` function body?

Reply:
Well spotted! Done that :)